analyzer/tests/functional/ctu_failure/test_ctu_failure.py

```diff
@@ -200,6 +275,62 @@ def check_source_in_archive(source_in_archive):
             check_source_in_archive("lib.c")
             check_source_in_archive("lib.h")
 
+    def test_ctu_on_demand_failure_zip_with_headers(self):
+        """
+        Test the failure zip contains the source of imported TU and all the
+        headers on which the TU depends when using on-demand mode.
+        """
+        self.__set_up_test_dir('ctu_failure_with_headers')
+        if not self.ctu_capable:
+            self.skipTest(NO_CTU_MESSAGE)
+        if not self.ctu_has_analyzer_display_ctu_progress:
+            self.skipTest(NO_DISPLAY_CTU_PROGRESS)
+        if not self.ctu_on_demand_capable:
+            self.skipTest(NO_CTU_ON_DEMAND_MESSAGE)
```

Comment:
This code is duplicated in multiple places. Can we create a separate helper function for it?